### ISSUE TITLE
Repaint issues in the tray area

### ISSUE THREAD

**Issue by ntrp** [polybar/polybar#1143]
I am running i3 and the Arc icon theme. I am using compton but the issue appears also without it.

The icons are fine on first paint (after started) and if they change because I muted the audio o disconnected from network they are completely broken:

![2018-04-05-194805_151x43_scrot](https://user-images.githubusercontent.com/720492/38382677-de2a3fb2-390a-11e8-84ce-cd521dcf39be.png)
![2018-04-05-194902_153x43_scrot](https://user-images.githubusercontent.com/720492/38382679-e0faf010-390a-11e8-8f6c-2b3e2f9071a6.png)

Polybar version
```
polybar 3.1.0-78-g224ffa9

Features: +alsa +curl +i3 -mpd +network +pulseaudio

X extensions: +randr (+monitors) -render -damage -sync -composite +xkb +xrm +xcursor

Build type: RelWithDebInfo
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -Wall -Wextra -Werror -Wno-noexcept-type -O2 -pedantic -pedantic-errors
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
```
Running Xorg and intel drivers.

**Comment by patrick96** [comment]
The tray is quite broken on some systems. We will have a rewrite of the tray at some point

**Comment by ntrp** [comment]
I see, too bad because it was working one week ago :(

**Comment by patrick96** [comment]
That's strange, we didn't change any tray code in a long time.

Any help is appreciated :)

**Comment by NBonaparte** [comment]
Might be caused by updating one or more libraries if it indeed worked before. If that is the case, then we can't do anything about it except roll back that library.

**Comment by bananflugan** [comment]
I have the same issue, the icons can randomly get broken. Another tray issue is that randomly when i start my computer or do changes to the configuration file some tray icons do not show up at all, its like the tray moves into a weird position which also causes my date module to not be fully visible.

Polybar 3.2.1

**Comment by leiserfg** [comment]
The same bug happens in i3bar and there you can "fix it" by using the transparent flag, so it is something fixable but I think related to a library update 'cause doesn't occurred before.

**Comment by patrick96** [comment]
Hmm, the latest development version cleaned up tray transparency a bit in #595, maybe this will help resolve this.

**Comment by JonnyHaystack** [comment]
> Hmm, the latest development version cleaned up tray transparency a bit in #595, maybe this will help resolve this.

So far after using master instead of the latest release it seems to be working a lot better for me :smiley:

**Comment by patrick96** [comment]
Awesome! I'm going to wait for a bit to hear from others if it's also fixed for them. Once 3.3.0 is released, most people should start to notice the change.

**Comment by JonnyHaystack** [comment]
After using it for longer, it seems it isn't completely fixed :cry: 
I don't seem to have the issues I used to have where different tray icons are repainted over each other, but sometimes the same application's icon is superimposed upon a previous state of that icon (e.g. as in @ntrp's screenshot).
I'm not sure why this is only sometimes occurring and I don't know what causes it. Nonetheless, it does seem to be an improvement. The tray will still need rewriting though, I suppose.

**Comment by patrick96** [comment]
Too bad, at least it got better.

> The tray will still need rewriting though, I suppose.

Yep, that was always the plan.

**Comment by patrick96** [comment]
Those of you who can reproduce this reliably. What icons cause it (from which application) and is it always the same icon?

Some time ago someone created a PR that supposedly fixes this, can you try out #1415 and report back if it fixed it?

**Comment by patrick96** [comment]
We have recently implemented a new tray that runs as a module where we rewrote a lot of the rendering code (#2609, #2689).

Some of these rendering issues may be fixed by this, but there are still some that may occur (I have been struggling to fix the dropbox icon using arbitrary images as its background data).
I am going to close this issue, if you experience rendering issues like these while using the new tray module (the old tray using `tray-position` still has the same issues), please report the application and your polybar version (`polybar -v`) together with a screenshot in #2933.
